src/utils/extensionParsing.ts

```diff
@@ -219,6 +222,62 @@ export function getEntityMetrics(typeIdx: number, extension: ExtensionStub, excl
   return [];
 }
 
+/**
+ * Extracts all metrics for a given entity with a condition that they match with
+ * metrics defined in the "metrics:" section of extension.yaml file and match not only
+ * by key but also by required dimensions, meaning that all of the required dimensions
+ * of the entity are contained within the metric defined in the "metrics:" section.
+ * Exclusions can be provided as a list of keys.
+ * @param typeIdx index of entity type within the topology section
+ * @param extension extension.yaml serialized as object
+ * @param excludeKeys keys to exclude from the response
+ * @returns list of matching metric objects
+ */
+export function getEntityMetricsMatchingDimensionsByRequiredDimensions(
```

Comment:
I don't quite like the duplication of `getEntityMetrics`  which already does most of this.